### ISSUE TITLE
Potential bug: VStar in environment resulting from bwdSlice

### ISSUE THREAD

**Issue by jstolarek** [jstolarek/slicer#46]
During backward slicing it is possible that `VStar` values are introduced into the returned environment. There's an extract function for dealing with this. I have not yet run into a situation where this would manifest, ie. where stars returned in the environment would cause an exception. I think this would only show as a resugaring exception. For now I'm leaving this unfixed until someone actually writes a program that hits this problem.

**Comment by jamescheney** [comment]
Yes, `extract` should be applied at the end to copy over the original values of any `VStar`s in the initial environment (or store).
